### DISCUSSION TITLE
Question on Restart / Restore Replica Command

### DISCUSSION THREAD

**Discussion by karthik-thiyagarajan** [ClickHouse/ClickHouse#53011] [Q&A]
Are the following two statements idem potent operation ?  Does it cause any problem, if I do this multiple times ? 

system restart replica on cluster `{cluster}` <table_name>
system restore replica <table_name> on cluster `{cluster}`

**Comment by melvynator** [comment]
They are used in two different context. 
SYSTEM RESTART will compare the current state with Zookeeper as a source of truth and add tasks to Zookeeper's queue if needed. It means that you need to have metadata in zookeeper and want to fix your local shards. 

RESTORE REPLICA 
Restores a replica if data is present but Zookeeper metadata is lost.

I don't see why someone would run this multiple times. These are generally used for troubleshooting reasons.
